### DISCUSSION TITLE
how to analysis the outputs?

### DISCUSSION THREAD

**Discussion by ItsAbba3** [mmschlk/shapiq#263] [Q&A]
hi any body knows how to analysis the outputs?

**Comment by mmschlk** [comment]
Hi @ItsAbba3, thank you very much for this question that currently is not answered at all in the repo haha. 😂 The following is quick attempt at answering this, but we plan on writing up a much more helpful guide on the do-s and don't-s of Shapley interaction interpretation.

The quick and oversimplifying answer is: the scores are Shapley values but also for groups of features and even groups of groups of features...

The interpretation also depends on the index you are using for computation. We define Shapley interactions (SIs) as interaction indices that if set to order 1, yield the Shapley value. Then the interpretation of the scores is the Shapley value. 

For example, the k-SII index set to order 2 will give you values for each interaction up to size 2. The first order values tell you how a feature individually influences your models prediction (positively/making the prediction higher, or negatively/making the prediction lower). The second order scores tell you a similar story. They tell you how your prediction is influenced by the group of features together but also taking into account how the two features already influence the prediction individually. Basically the second order scores tell you how much synergy features have and by how much the sum of two features is worth more than the worth of its two parts.


To make this a bit more concrete, consider the California housing dataset. Two important features are the latitude and longitude. Together both features encode the precise location of a property. They have a lot of synergy. The interaction is highly positive.

Some features have more interactions than others and even higher order interactions can play a role. Some features have no interaction or only main effects. 

For more information I can highly recommend the paper introducing the k-SII (https://arxiv.org/pdf/2209.04012) or maybe you already find our paper helpful (https://arxiv.org/abs/2410.01649, specifically the theoretical background). 
As said in the beginning stay tuned for more information or let's continue the discussion here.

**Comment by mmschlk** [comment]
To add to this; we recently written a first blog post regarding a high-level intro to Shapley interactions: We put this blog post into the [docs](https://shapiq.readthedocs.io/en/latest/introduction/index.html) titled "**What Are Shapley Interactions, and Why Should You Care?**".
